apps/confidential/confidential.go

```diff
@@ -207,6 +216,19 @@ func NewCredFromCertChain(certs []*x509.Certificate, key crypto.PrivateKey) (Cre
 	return cred, nil
 }
 
+// TokenProviderParameters is the authentication parameters passed to token providers
+type TokenProviderParameters = exported.TokenProviderParameters
+
+// TokenProviderResult is the authentication result returned by custom token providers
+type TokenProviderResult = exported.TokenProviderResult
+
+// NewCredFromTokenProvider creates a Credential from a function that provides access tokens. This
```

Comment:
For the public comments, please add a note that this extensibility point is only meant for Azure SDK to cache MSI tokens. We want to be explicit about this, so that other users do not try to use this. Long term we want to move all MSI logic into MSALs, so we could even drop this extensiblity point.